chapters/grid-search.qmd

```diff
@@ -0,0 +1,390 @@
+
+make_lhc <- function(ind, m, s) {
+  des <- DiceDesign::lhsDesign(s, m, seed = ind + 287)$design
+  colnames(des) <- paste("Parameter", letters[1:ncol(des)])
+  des <- as_tibble(des)
+  des$Replicate <- paste("Replicate", ind)
+
+  des
+}
+
+set.seed(202)
+map_dfr(1:3, ~ make_lhc(.x, 2, 10)) %>% 
+  ggplot(aes(`Parameter a`, `Parameter b`)) + 
+  geom_point(cex = 1.5) + 
+  coord_equal() +
+  geom_hline(yintercept = seq(0, 1, by = 1 / 10), alpha = 1 / 10) +
+  geom_vline(xintercept = seq(0, 1, by = 1 / 10), alpha = 1 / 10) +
+  lims(x = 0:1, y = 0:1) +
+  theme(panel.grid.minor = element_blank()) + 
+  facet_wrap(~ Replicate) +
+  theme(
+    panel.grid.major = element_blank(),
+    panel.grid.minor = element_blank(),
+  )
+```
+
+Technically, these designs cover the space of each predictor univariately. However, large multivariate regions can be empty. For example, one design only samples combinations along the diagonal. Additional constraints can make the design more consistent with our desires. 
+
+```{r}
+#| label: design-crit
+#| include: false
+
+rnd_maxmin <-
+  encode_set(bst_random %>% select(-grid), pset = bst_param, as_matrix = TRUE) %>% 
+  DiceDesign::mindist() %>% 
+  round(2)
+
+sfd_maxmin <-
+  encode_set(bst_sfd %>% select(-grid), pset = bst_param, as_matrix = TRUE) %>% 
+  DiceDesign::mindist() %>% 
+  round(2)
+
+reg_maxmin <-
+    encode_set(bst_reg %>% select(-grid), pset = bst_param, as_matrix = TRUE) %>% 
+    DiceDesign::mindist() %>% 
+    round(2)
+```
+
+For example, we could choose points to maximize the minimum pairwise distances between the candidates. These designs are usually referred to as MaxiMin designs [@pronzato2017minimax]. Comparing the two irregular designs in @fig-bst-grids, the random grid has a maximum minimum distance between candidates  of `r rnd_maxmin`. In contrast, the corresponding space-filling design's value (`r sfd_maxmin`) is `r round(sfd_maxmin/rnd_maxmin, 1)`-fold larger^[Using Euclidean distance.]. The latter design was optimized its coverage, and we can see far less redundancy in the optimized design. 
+
+A similar method, initially proposed by @audze1977new, maximizes a function of the inverse distances between $s$ candidate points: 
+
+$$
+criterion = \sum_{i=1}^s \sum_{j=1,\;i\ne j}^s\frac{1}{dist(\theta_i, \theta_j)^2}
+$$
+
+@bates2004formulation devised search methods to find optimal designs for this criterion.
+
+Some other space-filling designs of note: 
+
+- Maximum entropy sampling selects points based on assumptions related to the distributions of the tuning parameters and their covariance matrix [@shewry1987maximum;@joseph2015maximum]. 
+
+- Uniform designs [@fang2000uniform;@wang2022design] are designs that optimally allocate points so that they are uniformly distributed in the space. 
```

Comment:
```suggestion
- Uniform designs [@fang2000uniform;@wang2022design] optimally allocate points so that they are uniformly distributed in the space. 
```